src/gei/Commands/MigrateRepo/MigrateRepoCommandArgs.cs

```diff
@@ -79,10 +79,11 @@ public override void Validate(OctoLogger log)
                 {
                     throw new OctoshiftCliException("--ghes-api-url must be specified when --keep-archive is specified.");
                 }
-                if (UseGithubStorage)
-                {
-                    throw new OctoshiftCliException("--ghes-api-url must be specified when --use-github-storage is specified.");
-                }
+            }
+
+            if (UseGithubStorage && GhesApiUrl.IsNullOrWhiteSpace() && GitArchivePath.IsNullOrWhiteSpace())
+            {
+                throw new OctoshiftCliException("When using GitHub storage, you must provide --ghes-api-url, or --git-archive-path and --metadata-archive-path");
```

Comment:
Just a small suggestion here, as we are already grouping the ghes api url check in the above if block I would put this check also there:
```c#
if (GhesApiUrl.IsNullOrWhitespace())
{
    ...

    if (UseGitHubStorage && GitArchivePath.IsNullOrWhiteSpace())
    {
        ...
    }
}
```

Reply:
I was just testing an archive upload to AWS and realized that this check also needs to be added to `AwsBucketName` too:
```c#
if (AwsBucketName.HasValue() && GitArchivePath.IsNullOrWhiteSpace())
{
    ...
}
```

Reply:
Added your first suggestion in e7c36023ef2c35c43db2332bea812401c4044b82!

Reply:
Added your second suggestion in 6eb37d619def2f231167c0acbc8aa83966a3b4b5 :smile: